lib/Jobs/JobsList/JobsList.test.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { noop } from 'lodash';
+
+import {
+  screen, render,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+import '../../../test/jest/__new_mock__';
+
+import { JobsList } from '.';
+import { ItemFormatter } from '../tests/item-formatter';
+import { jobExecution } from '../../tests/jobExecution';
+
+const renderJobList = (
+  hasLoaded = true,
+  itemFormatter = noop,
+  isEmptyMessage = null,
+  jobs = []
+) => {
+  render(
+    <JobsList
+      jobs={jobs}
+      hasLoaded={hasLoaded}
+      itemFormatter={itemFormatter}
+      isEmptyMessage={isEmptyMessage}
+    />
+  );
+};
+
+describe('JobList', () => {
+  const emptyMessage = 'List is empty';
+
+  describe('rendering JobList with loaded state and empty list', () => {
+    it('should display JobList component', () => {
+      renderJobList(false, ItemFormatter, emptyMessage);
+
+      expect(screen.getByTestId('jobs-list-container')).toBeVisible();
+    });
+
+    it('should display preloader', () => {
+      renderJobList(false, ItemFormatter, emptyMessage);
+
+      expect(screen.getByTestId('preloader')).toBeVisible();
+    });
+
+    it('should hide jobs list', () => {
+      renderJobList(false, ItemFormatter, emptyMessage);
+
+      expect(screen.getByText(/stripes-components.endOfList/i)).toBeVisible();
+    });
+  });
+
+  describe('rendering JobList with loaded state and empty list', () => {
+    it('should display JobList component', () => {
+      renderJobList(true, ItemFormatter, emptyMessage);
```

Comment:
I suggest object in args, it's not clear what you pass - test becomes non-informative when test is kind of documentation

Reply:
> I suggest object in args, it's not clear what you pass - test becomes non-informative when test is kind of documentation

You mean ItemFormatter? It's required prop in that component. And also it's a util function for tests that return component related to received props.